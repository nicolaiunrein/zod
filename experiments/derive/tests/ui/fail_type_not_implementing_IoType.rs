mod test {
    struct X;

    #[derive(zod_derive_experiments::Zod)]
    struct Struct {
        pub value: u8,
        pub tuple: X,
    }

    #[derive(zod_derive_experiments::Zod)]
    struct Tuple(u8, X);
}

fn main() {}
