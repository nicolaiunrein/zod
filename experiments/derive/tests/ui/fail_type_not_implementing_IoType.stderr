error[E0277]: the trait bound `X: Type<Input>` is not satisfied
  --> tests/ui/fail_type_not_implementing_IoType.rs:13:20
   |
13 |         pub tuple: X,
   |                    ^ the trait `Type<Input>` is not implemented for `X`
   |
   = help: the following other types implement trait `Type<Io>`:
             <ConstStr<C, T> as Type<Input>>
             <ConstStr<C, T> as Type<experiments::Kind::Output>>
             <Struct as Type<Input>>
             <Struct as Type<experiments::Kind::Output>>
             <bool as Type<Input>>
             <bool as Type<experiments::Kind::Output>>
             <i16 as Type<Input>>
             <i16 as Type<experiments::Kind::Output>>
           and $N others

error[E0277]: the trait bound `X: Type<experiments::Kind::Output>` is not satisfied
  --> tests/ui/fail_type_not_implementing_IoType.rs:13:20
   |
13 |         pub tuple: X,
   |                    ^ the trait `Type<experiments::Kind::Output>` is not implemented for `X`
   |
   = help: the following other types implement trait `Type<Io>`:
             <ConstStr<C, T> as Type<Input>>
             <ConstStr<C, T> as Type<experiments::Kind::Output>>
             <Struct as Type<Input>>
             <Struct as Type<experiments::Kind::Output>>
             <bool as Type<Input>>
             <bool as Type<experiments::Kind::Output>>
             <i16 as Type<Input>>
             <i16 as Type<experiments::Kind::Output>>
           and $N others

error[E0277]: the trait bound `X: Type<Input>` is not satisfied
  --> tests/ui/fail_type_not_implementing_IoType.rs:18:22
   |
18 |     struct Tuple(u8, X);
   |                      ^ the trait `Type<Input>` is not implemented for `X`
   |
   = help: the following other types implement trait `Type<Io>`:
             <ConstStr<C, T> as Type<Input>>
             <ConstStr<C, T> as Type<experiments::Kind::Output>>
             <Struct as Type<Input>>
             <Struct as Type<experiments::Kind::Output>>
             <bool as Type<Input>>
             <bool as Type<experiments::Kind::Output>>
             <i16 as Type<Input>>
             <i16 as Type<experiments::Kind::Output>>
           and $N others

error[E0277]: the trait bound `X: Type<experiments::Kind::Output>` is not satisfied
  --> tests/ui/fail_type_not_implementing_IoType.rs:18:22
   |
18 |     struct Tuple(u8, X);
   |                      ^ the trait `Type<experiments::Kind::Output>` is not implemented for `X`
   |
   = help: the following other types implement trait `Type<Io>`:
             <ConstStr<C, T> as Type<Input>>
             <ConstStr<C, T> as Type<experiments::Kind::Output>>
             <Struct as Type<Input>>
             <Struct as Type<experiments::Kind::Output>>
             <bool as Type<Input>>
             <bool as Type<experiments::Kind::Output>>
             <i16 as Type<Input>>
             <i16 as Type<experiments::Kind::Output>>
           and $N others
